### ISSUE TITLE
Clarification on the data collection

### ISSUE THREAD

**Issue by gboeer** [AIMS/ozfish#2]
Hi and thanks for providing such a massive dataset. I have some questions regarding the data and I couldn't find any related publication, is there any additional resource available I may have missed? The only related ones I found were: https://apps.aims.gov.au/metadata/view/38c829d4-6b6d-44a1-9476-f9b0955ce0b8 and the data download page at https://data.pawsey.org.au/public/?path=/FDFML/labelled/manifests 

I try to express what I think so far and maybe somebody can clarify on that.
There are two kinds of annotations, the first one phrased 'Bounding Box Annotations' contains box coordinates and if any fish is present within or not. This was done using Amazon, so I'm thinking lots of people on Amazon Turk who judge if a given bounding box shows a fish or not? I guess those candidate boxes were generated using the code here at https://github.com/marrabld/open_fish_classifier .

The second dataset includes lots of species annotations and is called 'Crops' although here as well the bounding box coordinates are provided, and not only the cropped images. What becomes clear when looking at those images is, that only very few of the actually seen fishes per frame are annotated. In other words, it happens that one can see 10 fishes in an image clearly, however only 1 of them is annotated, and not even the best visible one. It would be good to know how those bounding boxes were chosen. Are they selected manually or again with some sort of detection algorithm?

Thanks for your contribution and best wishes.

**Comment by threehundred** [comment]
Hi @Legor, thanks for your questions, I'll try and clarify these for you.

The way the dataset is constructed is by-product of the way the videos are currently analysed by an ecologist. Our ecologists currently don't label videos and imagery in a way that feeds into an object detection algorithm. They are looking for the maximum number of each species of fish in the video, and the given frame for each particular species, called maxn. Once they have found maxn frames for each individual species, they then try and find the best frame where each fish is side on to the camera so they can measure the fish. 

The 'Crops' of fish are frames where a measurement of a fish was taken, because we are able to estimate a bounding box for extraction and have a species label for it. 

The 'Bounding Box Annotations'  are randomised images from all of the videos in this dataset that we have put to Amazon Ground Truth to be labelled as fish - there were two labellers per image. 

In doing so we were hoping to enable a two-stage detection pipeline, where something like YOLO can find and track fish. We could then use a secondary NN for doing the species classification. 

I hope this helps.

**Comment by gboeer** [comment]
Hi @threehundred, thanks for the clarification.
So are the Bounding Box Annotations done from scratch be the Amazon labellers or where they given pre-existing bounding boxes, and they just had to check if there is a fish or not? I'm asking because if they did it from scratch, the class 'no-fish' seems to be pretty arbitrarily.

One more thing, since you also use stereoscopic cameras, do you provide the calibration matrices for those as well?

Greetings

**Comment by threehundred** [comment]
Hi @Legor,

The labeling was done from scratch. There is only a single class, 0, which is for fish. 

We may be providing the calibration files through the BRII challenge, but not part of ozfish. The calibration and measurement is done with a proprietary piece of software called event measure, so we have kept that separate from this dataset.

**Comment by Hommus** [comment]
@Legor There are quite a few images with bounding boxes around coral or various underwater plants. Has this limited your ability to correctly detect fish or are they made redundant due to the large number of correct bounding boxes in the dataset?

**Comment by gboeer** [comment]
@Paulie-P Hi, yes I noted as well, that several of the annotations in the 'bounding-box' set are erroneous. This was the reason, why we didn't consider the OzFish dataset in its current state as a reliable benchmark dataset. Therefor I can't say how large the impact of those wrong annotations is regarding the training or evaluation performance of a detection model. I also don't have exact numbers how much of the annotations are wrong, so this assessment is based purely on my subjective observation, e.g. by looking at example images as below, where it can be seen, that only parts of the fish are annotated although the whole fish itself is annotated as well or that non-fish objects are annotated as fish.

![ozfish_example1](https://user-images.githubusercontent.com/1067159/93860536-86476380-fcbf-11ea-9666-94e431b4fea4.jpg)

![ozfish_example2](https://user-images.githubusercontent.com/1067159/93860849-f1913580-fcbf-11ea-83e9-69bc0b9814fc.jpg)

**Comment by threehundred** [comment]
Hi @Paulie-P @Legor, we had also noticed the same thing, and this dataset it yet to be fully cleaned, we've taken the results directly from MTurk. In terms of fish detection we've found it to still yield very good results, with the odd detection of habitat in the background. 

In terms of real world datasets though, noisiness is to be expected (for both data and labels), especially when dealing with underwater imagery.

**Comment by Hommus** [comment]
@threehundred I'm using the OzFish dataset on my undergrad thesis and it's been fun so far. What kind of results were you getting for detection, in terms of mAP or other benchmarks you've been using? It'll give me something to aim towards! I've been running experiments for a few months now and the largest increase (in mAP) has come from increasing the resolution of the network (416x416 to 608x608). I haven't finished training it yet, but it's currently at ~70% mAP.

**Comment by BenMaslen** [comment]
@threehundred, have there been any updates on possible fixes to the issues described by @Legor? I am running some Yolo models on this data set at the moment and at the stage it seems that of the 42,685 labels, the uninformative 'fish' label was given 35,694 times.

**Comment by BenMaslen** [comment]
@Hommus 70% is better then me so far on this data set! What object detection algorithm's have you been running? Did you convert all the labels to 'fish' for identiication instead of classification?

**Comment by threehundred** [comment]
Hi @BenMaslen we've done no further cleaning - but happy to take updates from the community.

In terms of having 'fish' labels on the boxes, this is the best we can do at the moment, as identifying species on still frames is not how the videos are annotated in practice (usually done by watching movement in video - not single frames). 

The best approach we've found so far with this dataset is to use a two stage approach 1) Fish detection using object detector trained with bbox dataset 2) FCN trained on the fish crops for species identification. 

Hope that helps.

**Comment by BenMaslen** [comment]
Hi @threehundred thankyou for the clarification!

**Comment by Hommus** [comment]
@BenMaslen YOLOv4 with images at 608x608 resolution and a learning rate of 0.001. I also had the duration of the experiment at 6000 "iterations" in my report, which I assume refers to epochs. 

The experiment was run on the fish/no-fish bounding box annotations, so there was no conversion. I was following the two-stage method that @threehundred mentioned.
